user_modules/shared_testcases/testCasesForPerformAudioPassThru.lua

```diff
@@ -0,0 +1,566 @@
+        ttsChunks =
+        {
+
+          {
+            text = ttsChunksType.text,
+            type = ttsChunksType.type
+          },
+        }
+
+      })
+    :Do(function(_,data)
+
+        self.hmiConnection:SendResponse(data.id, "TTS.Speak", "UNSUPPORTED_RESOURCE", { })
+
+      end)
+
+    --hmi side: expect for UI.PerformAudioPassThru
+    EXPECT_HMICALL("UI.PerformAudioPassThru",
+      {
+        appID = self.applications[applicationName],
+        audioPassThruDisplayTexts = {
+          {fieldName = "audioPassThruDisplayText1", fieldText = "DisplayText1"},
+          {fieldName = "audioPassThruDisplayText2", fieldText = "DisplayText2"},
+        },
+        maxDuration = 2000,
+        muteAudio = true,
+        audioPassThruIcon = { imageType = "STATIC", value = "icon.png"}
+      })
+    :Do(function(_,data)
+        self.hmiConnection:SendResponse(data.id, "UI.PerformAudioPassThru", "SUCCESS", {})
+      end)
+
+    self.mobileSession:ExpectResponse(CorIdPerformAudioPassThruSpeechCap, { success = true, resultCode = "WARNINGS"})
+  else 
+  	print ("Unexpected TTS.Chunks type")
+  end
+end
+
+function testCasesForPerformAudioPassThru:PerformAudioPassThru_DYNAMIC_image_SUCCESS_all(self)
+  local CorIdPerformAudioPassThruAppParVD= self.mobileSession:SendRPC("PerformAudioPassThru",
+    {
+      initialPrompt =
+      {
+        {
+          text = "Makeyourchoice",
+          type = "TEXT",
+        },
+
+      },
+      audioPassThruDisplayText1 = "DisplayText1",
+      audioPassThruDisplayText2 = "DisplayText2",
+      samplingRate = "16KHZ",
+      maxDuration = 2000,
+      bitsPerSample = "8_BIT",
+      audioType = "PCM",
+      muteAudio = true,
+      audioPassThruIcon =
+      { value = "icon.png",
+        imageType = "DYNAMIC"
+      }
+    })
+
+  -- hmi expects TTS.Speak request
+  EXPECT_HMICALL("TTS.Speak",
+    {
+      speakType = "AUDIO_PASS_THRU",
+      ttsChunks = { { text = "Makeyourchoice", type = "TEXT" } },
+      appID = self.applications[applicationName]
+    })
+  :Do(function(_,data)
+      -- send notification to start TTS.Speak
+      self.hmiConnection:SendNotification("TTS.Started",{ })
+
+      -- HMI sends TTS.Speak SUCCESS
+      local function ttsSpeakResponse()
+        self.hmiConnection:SendResponse (data.id, data.method, "SUCCESS", {})
+
+        -- HMI sends TTS.Stop
+        self.hmiConnection:SendNotification("TTS.Stopped")
+      end
+
+      RUN_AFTER(ttsSpeakResponse, 1000)
+    end)
+
+  -- hmi expects UI.PerformAudioPassThru request
+  EXPECT_HMICALL("UI.PerformAudioPassThru",
+    {
+      appID = self.applications[applicationName],
+      audioPassThruDisplayTexts = {
+        {fieldName = "audioPassThruDisplayText1", fieldText = "DisplayText1"},
+        {fieldName = "audioPassThruDisplayText2", fieldText = "DisplayText2"},
+      },
+      maxDuration = 2000,
+      muteAudio = true,
+      audioPassThruIcon = { imageType = "DYNAMIC", value = PathToAppFolder .."icon.png"}
```

Comment:
@maria-mih , please make verification of 'value' to be checked as sub-string, because SDL can return as result path correct path but declared in other string way and ATF will return error.
You can use as example:
if (string.match(data.params.cmdIcon.value, "%S*" .. "("..string.sub(storagePath, 2).."icon.png)" .. "$") == nil ) then
-- print some error
end